src/tests/roc_sndio/target_sox/test_player_recorder.cpp

```diff
@@ -115,41 +115,60 @@ TEST_GROUP(player_recorder) {
 };
 
 TEST(player_recorder, player_noop) {
-    MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 }
 
 TEST(player_recorder, player_error) {
-    MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     CHECK(!player.open("/bad/file"));
 }
 
 TEST(player_recorder, player_start_stop) {
     MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     core::TempFile file("test.wav");
     CHECK(player.open(file.path()));
 
-    player.start();
+    player.start(receiver);
     player.stop();
     player.join();
 }
 
 TEST(player_recorder, player_stop_start) {
     MockReceiver receiver;
-    Player player(receiver, buffer_pool, allocator, true, ChMask, SampleRate);
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
 
     core::TempFile file("test.wav");
     CHECK(player.open(file.path()));
 
     player.stop();
-    player.start();
+    player.start(receiver);
     player.join();
 }
 
+TEST(player_recorder, player_open_zero_sample_rate) {
+    Player player(buffer_pool, allocator, true, ChMask, 0);
+
+    core::TempFile file("test.wav");
+    CHECK(player.open(file.path()));
+    CHECK(player.get_sample_rate() != 0);
+}
+
+TEST(player_recorder, player_open_non_zero_sample_rate) {
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate);
+
+    core::TempFile file("test.wav");
+    CHECK(player.open(file.path()));
+    CHECK(player.get_sample_rate() == SampleRate);
+}
+
+TEST(player_recorder, player_open_sample_rates_mistmatch) {
+    Player player(buffer_pool, allocator, true, ChMask, SampleRate + 1);
+    CHECK(!player.open(NULL, NULL));
```

Comment:
There are three problems with this test:

* Opening an audio device may have side-effects. For example, if the user is using ALSA without dmix, only one process can open an audio device. So this test will either fail to open it (if it's already opened by someone else), or prevent other processes from opening it for a short time.

* If this test will fail to open device for some other reason (e.g. an ALSA device is already opened by someone else), it will pass anyway, so we really don't known if it actually tests something.

* I'm not sure that any SoX backend is guaranteed to fail with rate 44100+1.

We theoretically may want a separate set of tests that work with a real audio device. Such tests should not be run automatically, but instead only when the user explicitly asked that. However I'm not sure it's worth it.

Thus, if we can't cover this case with a test that uses a temporary file instead of real audio device, I suggest to remove it for now.

Reply:
I think that you've reviewed an old version of tests, I've rebased recently.

Reply:
The first two points are still relevant after rebase.

Reply:
Fixed